upload-api/tables/index.js

```diff
@@ -98,4 +98,25 @@ export const rateLimitTableProps = {
   globalIndexes: {
     subject: { partitionKey: 'subject', projection: ['rate', 'id'] },
   }
+}
+
+/**
+ * Track revocations.
+ * 
+ * This table is designed to be batch-GET-queried by delegation CIDs,
+ * (which means the primary key MUST be just the delegation CID) but needs to accomodate
+ * multiple possible "revocation context CIDs" each with its own "cause CID". Because
+ * BatchGetCommand only works on primary tables, not indices, we need to cram
+ * the context CID and the cause CIDs into a "set" field of :-separated contextCID:causeCID
+ * strings.
+ * 
+ * @type TableProps 
+*/
+export const revocationTableProps = {
+  fields: {
+    delegation: 'string',     // `baf...x`(CID of the revoked delegation)
+    // @ts-ignore SST seems to incorrectly limit the set of field types to the three types suitable for index keys - hopefully this is fixed in the next versoin
+    contextsAndCauses: 'set'  // a set of :-separated contextCID:causeCID pairs
```

Comment:
should we consider using serialized JSON instead so it's possible to upgrade, I don't know if it does but thought I'd ask so tradeoffs could be assesed.

Reply:
serialized JSON seems fine yep! though I'll probably use CBOR here because (answering your second question as well) the max size of any item in Dynamo is 400KB (https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/ServiceQuotas.html#limits-items) 

unfortunately I don't think `BS` is workable - the reason I'm using a `set` here is so that we can do an atomic update using the `ADD` update expression you noticed below - if we use just about anything else we'll need to fetch the item before updating it, which imho is a dealbreaker

Reply:
worth noting - we might be able to use a `list` or `map` type for this column, but I don't think they would be more compact or efficient than `set` and I have not tried to get them working.

also, after thinking about this, I'm not sure serialized JSON or CBOR would be more compact than the `:`-separated value I'm using now, so I think I'd prefer to stick with what I have for now... open to being convinced otherwise, but I'd like to optimize for the smallest set values possible!